test/services/companyPscService.unit.ts

```diff
@@ -0,0 +1,99 @@
+import { createOAuthApiClient } from "../../src/services/apiClientService";
+import { getCompanyIndividualPscList, getCompanyPscList } from "../../src/services/companyPscService";
+import { COMPANY_NUMBER, EMPTY_COMPANY_PSC_LIST, VALID_COMPANY_PSC_LIST } from "../mocks/companyPsc.mock";
+
+jest.mock("@companieshouse/api-sdk-node");
+jest.mock("../../src/services/apiClientService");
+
+const mockGetCompanyPsc = jest.fn();
+const mockCreateOAuthApiClient = createOAuthApiClient as jest.Mock;
+let session: Session;
+
+mockCreateOAuthApiClient.mockReturnValue({
+    companyPsc: {
+        getCompanyPsc: mockGetCompanyPsc
+    }
+});
+
+describe("companyPscService", () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+        session = new Session();
+    });
+
+    it("getCompanyPscList should return 200 OK HttpStatus response", async () => {
+        const mockResponse: ApiResponse<CompanyPersonsWithSignificantControlResource> = {
+            httpStatusCode: HttpStatusCode.Ok,
+            resource: VALID_COMPANY_PSC_LIST
+        };
+        mockGetCompanyPsc.mockResolvedValueOnce(mockResponse as ApiResponse<CompanyPersonsWithSignificantControlResource>);
+        const request = {} as Request;
+        // request.query = { companyNumber: COMPANY_NUMBER };
+
+        const response = await getCompanyPscList(request, COMPANY_NUMBER);
+
+        const resource = response.resource as CompanyPersonsWithSignificantControlResource;
+        expect(response.httpStatusCode).toBe(HttpStatusCode.Ok);
+        expect(resource).toEqual(VALID_COMPANY_PSC_LIST);
+        expect(mockCreateOAuthApiClient).toHaveBeenCalledTimes(1);
+        expect(mockGetCompanyPsc).toHaveBeenCalledTimes(1);
+        expect(mockGetCompanyPsc).toHaveBeenCalledWith(COMPANY_NUMBER);
+
+    });
+    it("getCompanyPscList should throw an error if HttpStatus is not 200 OK", async () => {
+        const mockResponse: ApiResponse<CompanyPersonsWithSignificantControlResource> = {
+            httpStatusCode: HttpStatusCode.ServiceUnavailable
+        };
+        mockGetCompanyPsc.mockResolvedValueOnce(mockResponse as ApiResponse<CompanyPersonsWithSignificantControlResource>);
+        const request = {} as Request;
+
+        try {
+            const response = await getCompanyPscList(request, COMPANY_NUMBER);
+        } catch (error: any) {
+            expect(error.message).toBe("getCompanyPscList - Failed to get company psc list for company number 12345678");
+        }
+    });
+    it("getCompanyPscList should throw an error if no resource is returned", async () => {
+        const mockResponse: ApiResponse<CompanyPersonsWithSignificantControlResource> = {
+            httpStatusCode: HttpStatusCode.Ok
+        };
+        mockGetCompanyPsc.mockResolvedValueOnce(mockResponse as ApiResponse<CompanyPersonsWithSignificantControlResource>);
+        const request = {} as Request;
+        try {
+            await getCompanyPscList(request, COMPANY_NUMBER);
```

Comment:
Again fail if no error thrown?

Reply:
After line 68, we can add:

`throw new Error("did not get here");`